### PR TITLE
GEODE-9052: rename the geode-redis module

### PR DESCRIPTION
**Pull request by nonbinaryprogrammer**
rename the geode-redis module to geode-apis-compatible-with-redis. this
is so that this name doesn't end up in a customer-facing location that
may cause us to violate Redis's copyright

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Review by onichols-pivotal — APPROVED** [review]

File: geode-redis/src/acceptanceTest/resources/0001-configure-redis-tests.patch [hunk 1]
```diff
@@ -1,808 +0,0 @@
-From 15b752b4d129d4f303d033f0e6e63871355388cc Mon Sep 17 00:00:00 2001
```
Comment (sabbey37) [inline geode-redis/src/acceptanceTest/resources/0001-configure-redis-tests.patch:1]: This file needs to be added back.  If you look at the output from the `execute_tests` task of the RedisTests in the pipeline, you'll see:
```
grep: geode/geode-redis/build.gradle: No such file or directory

```

**Review by sabbey37 — CHANGES_REQUESTED** [review]
Looks like the `GeodeDependencyJarIntegrationTest > verifyManifestClassPath` also needs to be updated.  It would be comforting to see the Integration, DUnit, and Acceptance tests run properly and pass as well, though I imagine they'll be fine after that test is fixed.

**Comment by sabbey37** [comment]
This line in `execute_tests.sh` also needs to be updated:
```
if [ "${GRADLE_TASK}" = ":geode-redis:redisAPITest" ] && ! grep -q redisAPITest geode/geode-redis/build.gradle ; then
```

Right now the RedisTest is printing: 
```
redisAPITest is not applicable for 1.13 and earlier
```
because it is still looking for the redisAPITest task in the `geode-redis` module

**Review by sabbey37 — APPROVED** [review]
Looks good! Good call on all the other changes!  I'm realizing the concourse pipeline won't run with the updated task until after it's been merged.  I'm not sure if it needs to be manually re-fly-ed, but @onichols-pivotal would know.

**Comment by onichols-pivotal** [comment]
> I'm not sure if it needs to be manually re-fly-ed.
no manual involvement needed, the geode pipeline has been self-reflying since Dec 2017.

**Comment by sabbey37** [comment]
> > I'm not sure if it needs to be manually re-fly-ed.
> > no manual involvement needed, the geode pipeline has been self-reflying since Dec 2017.

😲 🚀

**Review by rhoughton-pivot — APPROVED** [review]

**Pull request merged**